### ISSUE TITLE
Upgrade smallvec to 0.6

### ISSUE THREAD

**Issue by gnzlbg** [integer32llc/rust-playground#249]
One should probably upgrade all crates, but this will probably prevent many permalinks that are using incompatible crate versions from compiling.

**Comment by shepmaster** [comment]
> many permalinks that are using incompatible crate versions from compiling.

Can you please provide an example of the problem?

**Comment by gnzlbg** [comment]
If I create a perma-link to the integer 32 playground to an example using:

```rust
extern crate smallvec;

fn main() {
   let v = smallvec::SmallVec::with_capacity(100);
}
```

is the version of the `smallvec` crate used registered as part of the link or in the integer32 play ground somewhere?

If the `smallvec` crate is upgraded to a newer version that changes the major version number (such that older APIs are potentially broken after the upgrade), will old links continue to use older versions of the crate?

**Comment by shepmaster** [comment]
> is the version of the smallvec crate used registered as part of the link or in the integer32 play ground somewhere?

> will old links continue to use older versions of the crate?

No.

See also:
- https://github.com/integer32llc/rust-playground/issues/101#issuecomment-282899611
- https://github.com/integer32llc/rust-playground/issues/101#issuecomment-282908899

You are essentially asking for a playground with every possible subset and combination of every (Rust version, crate, crate version) tuple. As you might understand, that's not currently temporally / financially feasible.

**Comment by gnzlbg** [comment]
I was just asking for smallvec to be upgraded to its latest version, and saying that it might make sense to upgrade all crates to their latest version too, while recognizing that this might break many links to the playground. 

But as you mention, the breakage issue is not really a problem, since it has no current solution (maybe someday).

**Comment by shepmaster** [comment]
> while recognizing that this might break many links to the playground.

Ah, gotcha. Your original wording confused me, so I wasn't really clear on why you were bringing it up.

**Comment by shepmaster** [comment]
Closed via #250

**Comment by gnzlbg** [comment]
Thanks!
